thor/main.py

```diff
@@ -137,42 +137,121 @@ def clusterVelocity(
         If clusters are found, will return a list of numpy arrays containing the 
         observation IDs for each cluster. If no clusters are found, will return np.NaN.
     """ 
+    logger.debug(f"cluster: vx={vx} vy={vy} n_obs={len(obs_ids)}")
     xx = x - vx * dt
     yy = y - vy * dt
+
+    X = np.stack((xx, yy), 1)
+
+    clusters = find_clusters(X, eps, min_samples, alg=alg)
+    clusters = filter_clusters_by_length(clusters, dt, min_samples, min_arc_length)
+
+    cluster_ids = []
+    for cluster in clusters:
+        cluster_ids.append(obs_ids[cluster])
```

Comment:
This is a little different than the previous code, which used the `db.labels_` fitting results inline, directly inside a `for cluster in clusters:` loop. The result was that the code only needed to iterate over `clusters` once, while mine needs to do that twice.

But I think that's OK, because `clusters` is generally going to be a very small list.

Reply:
Looks good to me! I like splitting up the filtering into a different function, it's much cleaner and more maintainable.